src/jit/importer.cpp

```diff
@@ -2878,24 +2878,36 @@ GenTreePtr      Compiler::impIntrinsic(CORINFO_CLASS_HANDLE     clsHnd,
                                        CORINFO_SIG_INFO *       sig,
                                        int                      memberRef,
                                        bool                     readonlyCall,
-                                       CorInfoIntrinsics *      pIntrinsicID)
+                                       CorInfoIntrinsics *      pIntrinsicID,
+                                       bool *                   pMustExpand)
 {
-    CorInfoIntrinsics intrinsicID = info.compCompHnd->getIntrinsicID(method);
+    CorInfoIntrinsics rawIntrinsicID = info.compCompHnd->getIntrinsicID(method);
```

Comment:
getIntrinsicID() is called by JIT in many places (see gentree.cpp, morph.cpp even multiple times in importer.cpp).  JIt performs checks like

if (info.compCompHnd->getIntrinsicID(tree->gtCall.gtCallMethHnd) == CORINFO_INTRINSIC_Object_GetType) {..}

We should add JIT wrapper method for this that returns two bits original intrinsicID and a flag whether it must be expanded and replace all the calls with JIT wrapper.

It will also avoid the need for code on lines 2897-2900